grype-server/pkg/scanner/scanner.go

```diff
@@ -98,11 +100,16 @@ func (s *Scanner) ScanSbomJson(sbom string) (*models.Document, error) {
 	if err != nil {
 		return nil, fmt.Errorf("unable to decode sbom: %v", err)
 	}
-	if formatOption == format.UnknownFormatOption {
-		return nil, fmt.Errorf("unknown SBOM format option: %v", formatOption)
+
+	// Validate returns error despite the syftSbom is valid so just log error and continue
+	if err := formatOption.Validate(sbomReader); err != nil {
+		log.Errorf("unknown SBOM format option: %v, %v", formatOption.ID().String(), err)
+	}
+	if syftSbom.Artifacts.PackageCatalog == nil {
+		return nil, fmt.Errorf("packagecatalog is empty")
 	}
 
-	packages := grype_pkg.FromCatalog(syftSbom.Artifacts.PackageCatalog)
+	packages := grype_pkg.FromCatalog(syftSbom.Artifacts.PackageCatalog, grype_pkg.ProviderConfig{})
```

Comment:
is it good to use an empty `grype_pkg.ProviderConfig{}`?

Reply:
The only option that is used by `FromCatalog` is `GenerateMissingCPEs` that I forgot to set So I will set it true.
https://github.com/anchore/grype/blob/f094b860b9b9ebcdf06d9b9c6f7d5c000fad395b/grype/pkg/package.go#L66-L67